x/evm/client/rest/rest.go

```diff
@@ -1,90 +1,89 @@
-	r.HandleFunc("/txs/decode", authrest.DecodeTxRequestHandlerFn(cliCtx)).Methods("POST") // default from auth
+// RegisterTxRoutes - Central function to define routes that get registered by the main application
+func RegisterTxRoutes(clientCtx client.Context, rtr *mux.Router) {
+	r := clientrest.WithHTTPDeprecationHeaders(rtr)
+	r.HandleFunc("/txs/{hash}", QueryTxRequestHandlerFn(clientCtx)).Methods("GET")
+	r.HandleFunc("/txs", authrest.QueryTxsRequestHandlerFn(clientCtx)).Methods("GET")
+	r.HandleFunc("/txs", authrest.BroadcastTxRequest(clientCtx)).Methods("POST")
+	r.HandleFunc("/txs/encode", authrest.EncodeTxRequestHandlerFn(clientCtx)).Methods("POST")
+	r.HandleFunc("/txs/decode", authrest.DecodeTxRequestHandlerFn(clientCtx)).Methods("POST")
 }
 
-func QueryTxRequestHandlerFn(cliCtx context.CLIContext) http.HandlerFunc {
+// QueryTxRequestHandlerFn implements a REST handler that queries a transaction
+// by hash in a committed block.
+func QueryTxRequestHandlerFn(clientCtx client.Context) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		vars := mux.Vars(r)
 		hashHexStr := vars["hash"]
 
-		cliCtx, ok := rest.ParseQueryHeightOrReturnBadRequest(w, cliCtx, r)
+		clientCtx, ok := rest.ParseQueryHeightOrReturnBadRequest(w, clientCtx, r)
 		if !ok {
 			return
 		}
 
 		ethHashPrefix := "0x"
-		if strings.HasPrefix(hashHexStr, ethHashPrefix) {
-			// eth Tx
-			ethHashPrefixLength := len(ethHashPrefix)
-			output, err := getEthTransactionByHash(cliCtx, hashHexStr[ethHashPrefixLength:])
-			if err != nil {
-				rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
-				return
-			}
-			rest.PostProcessResponseBare(w, cliCtx, output)
+		if !strings.HasPrefix(hashHexStr, ethHashPrefix) {
+			authrest.QueryTxRequestHandlerFn(clientCtx)
 			return
 		}
 
-		output, err := utils.QueryTx(cliCtx, hashHexStr)
+		// eth Tx
+		ethHashPrefixLength := len(ethHashPrefix)
+		output, err := getEthTransactionByHash(clientCtx, hashHexStr[ethHashPrefixLength:])
 		if err != nil {
-			if strings.Contains(err.Error(), hashHexStr) {
-				rest.WriteErrorResponse(w, http.StatusNotFound, err.Error())
-				return
-			}
 			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
 			return
 		}
-		rest.PostProcessResponseBare(w, cliCtx, output)
-	}
 
+		rest.PostProcessResponseBare(w, clientCtx, output)
+	}
 }
 
 // GetTransactionByHash returns the transaction identified by hash.
-func getEthTransactionByHash(cliCtx context.CLIContext, hashHex string) ([]byte, error) {
+func getEthTransactionByHash(clientCtx client.Context, hashHex string) ([]byte, error) {
 	hash, err := hex.DecodeString(hashHex)
 	if err != nil {
 		return nil, err
 	}
-	node, err := cliCtx.GetNode()
+	node, err := clientCtx.GetNode()
 	if err != nil {
 		return nil, err
 	}
-	tx, err := node.Tx(hash, false)
+
+	tx, err := node.Tx(context.Background(), hash, false)
```

Comment:
im having some trouble understanding what an emptyCtx is used for here for `context.Background()`